pkg/controller/gitlab/buckets.go

```diff
@@ -64,6 +150,9 @@ func (r *bucketReconciler) reconcile(ctx context.Context) error {
 	}
 
 	r.status = &bucket.Status
+	if bucket.Status.IsReady() {
+		return r.secretTransformer.transform(ctx)
```

Comment:
will this transform function continue to be called everytime that the bucket is in the ready status? is there a short-circuit to skip doing it after it's been done once?

Reply:
We may not need to short circuit anything since I was planning to use the default `syncPeriod` value, i.e. `10h`. Under that condition, the GitLab object should not be `requeued` unless it has been modified (externally).